src/Build/BackEnd/Components/Logging/LoggingContext.cs

```diff
@@ -309,8 +309,17 @@ internal void LogFatalBuildError(Exception exception, BuildEventFileInfo file)
         /// <param name="filePath">Path to response file</param>
         internal void LogIncludeFile(string filePath)
         {
-            ErrorUtilities.VerifyThrow(IsValid, "must be valid");
+            CheckValidity();
             _loggingService.LogIncludeFile(BuildEventContext, filePath);
         }
+
+        private protected void CheckValidity()
+        {
+            if (!_isValid)
+            {
+                ErrorUtilities.VerifyThrow(_isValid, "LoggingContext (type: {0}) was not valid during logging attempt.",
+                    this.GetType());
+            }
```

Comment:
can be simplified since ErrorUtilities.VerifyThrow already checks the value of _isValid.
```suggestion

            ErrorUtilities.VerifyThrow(_isValid, "LoggingContext (type: {0}) was not valid during logging attempt.",
                this.GetType());
```

Reply:
This requires always materializing the string before the call (we don't have interpolated string handlers in our ErrorUtilities at the moment), so I'd fix it the other way

```suggestion
            if (!_isValid)
            {
                ErrorUtilities.ThrowInternalError("LoggingContext (type: {0}) was not valid during logging attempt.",
                    this.GetType());
            }
```